File: Lecture013 Binary Search Interview Questions/Peak_in_Montain_Array.swift
func peakIndexInMountainArray(arr: [Int]) -> Int {
   var start = 0
   var end = arr.count - 1
   var mid = start + (end - start) / 2

   while start < end {
       if arr[mid] < arr[mid + 1] {
           start = mid + 1
       } else {
           end = mid
       }
       mid = start + (end - start) / 2
   }
   return start
}

// Example usage
let arr = [1, 2, 3, 4, 5, 4, 3, 2, 1]
let peakIndex = peakIndexInMountainArray(arr: arr)
print("Peak index in the array: \(peakIndex)")

File: 013_Binary_Search_Interview_Questions/Peak_in_Montain_Array.py
from ast import List

def peakIndexInMountainArray(arr: int) -> int:
    start = 0
    end = len(arr) - 1

    while start < end:
        mid = start + (end - start) // 2

        if arr[mid] < arr[mid + 1]:
            start = mid + 1
        else:
            end = mid

    return start

# Example usage
arr = [1, 2, 3, 4, 5, 4, 3, 2, 1]
peak_index = peakIndexInMountainArray(arr)
print("Peak index in the array:", peak_index)